### ISSUE TITLE
Adding Takahe contacts failing

### ISSUE THREAD

**Issue by mexon** [friendica/friendica#14175]
### Bug Description

I am unable to follow vote@fedi.vision as an ActivityPub contact.  It turns up as an RSS feed instead.  That defeats the purpose of this particular bot - you can't send your vote to it if it's RSS.

### Steps to Reproduce

1. Click "contact" link
2. Enter "vote@fedi.vision" into "Add New Contact" box
3. Click "connect"
4. On "Connect/Follow" page, click "Submit Request"
5. Contact is now followed as an RSS feed

Actual Result:

Contact is followed as an RSS feed

Expected Result:

Contact is followed as an ActivityPub contact

### Platform Info

Friendica Version: 2024.03
Friendica Source: git
PHP version: 8.1
SQL version: 10.1

### Additional Info

Issue seems to be at [this line](https://github.com/friendica/friendica/blob/d20c5493ae4f2967cf01c3e33b11683a3a7bc36f/src/Model/APContact.php#L161).  Takahe happens to have profile URLs that contain at signs, like https://fedi.vision/@vote@fedi.vision/.  This triggers the workaround for Mastodon, which doesn't appear to work.  Adding a hack to bypass the Mastodon code for a particular site seems to fix the problem.

**Comment by mexon** [comment]
I believe caused by #13359

**Comment by annando** [comment]
Can you go to the admin section and do a probe there?

**Comment by mexon** [comment]
With a "clean" codebase, i.e. my local hack stashed, it seems to have a network of `apub`, so I guess that works fine:
```
Array
(
    [name] => FediVision Announcements
    [given_name] => 
    [family_name] => 
    [nick] => announce
    [guid] => 
    [url] => https://fedi.vision/@announce@fedi.vision/
    [addr] => announce@fedi.vision
    [alias] => https://fedi.vision/@announce/
    [photo] => https://eu-central-1.linodeobjects.com:443/st4/profile_images/2024/1/22/HHhRgKaREKwsx2U8dVkNrHYKcI0.jpg
    [photo_medium] => 
    [photo_small] => 
    [header] => https://eu-central-1.linodeobjects.com:443/st4/background_images/2024/1/22/sQSNPk4fetZV7-KEcK9xmHz50VI.jpg
    [account-type] => 0
    [community] => 
    [keywords] => 
    [location] => 
    [about] => Announcements about the FediVision song contest, running in May 2024! Please follow for dates and news. See [url=https://fedivision.party/]https://fedivision.party/[/url] for rules and eligibility.

Posts by @TQ@weirder.earth, @zatnosk@mastodon.art, @futzle@old.mermaid.town.
    [xmpp] => 
    [matrix] => 
    [hide] => 
    [batch] => https://fedi.vision/inbox/
    [notify] => https://fedi.vision/@announce@fedi.vision/inbox/
    [poll] => https://fedi.vision/@announce@fedi.vision/outbox/
    [request] => 
    [confirm] => 
    [subscribe] => 
    [poco] => 
    [following] => https://fedi.vision/@announce@fedi.vision/following/
    [followers] => https://fedi.vision/@announce@fedi.vision/followers/
    [inbox] => https://fedi.vision/@announce@fedi.vision/inbox/
    [outbox] => https://fedi.vision/@announce@fedi.vision/outbox/
    [sharedinbox] => https://fedi.vision/inbox/
    [priority] => 0
    [network] => apub
    [pubkey] => -----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAw/dbG5+thlFoWjsRMHt2
fhd/YQf3U0RfUQ+wQMTIEONLU5m97pKtQbgAIick7b/7dISWuJ0etMsU+HtHbr0F
z+LzPCQXUYPRFQ+7vDgSvbKi9A2QOBhZQdMmeEDf7mkOp8aJ2bD2ClQKMgPvtzQs
Gv3f/BGtBITmNVo8p5srsALpeOlg1PeThEiJ2epx211SnEJWb2CT9Lcklq7Q5d39
DZKyPO0sFrapq/jf5GSkOtONsfEM1YK+Rk82HTF5ai8TwCWf3jG1BPqjTF1Kf9t8
qFHwCY+OGWMojqspqk1Dv0mgDdJliDLQOKAzVGMDAbGdVH33lCta7LIVff7E2rqP
4QIDAQAB
-----END PUBLIC KEY-----
    [manually-approve] => 
    [baseurl] => http://fedi.vision
    [gsid] => 13789
    [networks] => Array
        (
        )

)
```

**Comment by foss-** [comment]
Following `vote@fedi.vision` via the add contact box worked fine on This is Friendica, version 2024.06-dev that is running at the web location https://pirati.ca. The database version is 1561/1561, the post update version is 1550/1550.

Probably indeed fixed in https://github.com/friendica/friendica/pull/13492 addressing https://github.com/friendica/friendica/issues/13359

**Comment by annando** [comment]
@foss- no, it must be unrelated to https://github.com/friendica/friendica/pull/13492, since that PR is part of 2024.03, the version that @mexon is using.

**Comment by sekretaerbaer** [comment]
I have tested this and I can only follow `vote@fedi.vision `as RSS. 
I have the same with the account `Sebastian@Pertsch.social `
version 2024.06-dev

**Comment by foss-** [comment]
Can confirm that I can neither follow `@Sebastian@Pertsch.social`  nor `Sebastian@Pertsch.social` with friendica throwing error `The network type couldn't be detected. Contact can't be added.` on This is Friendica, version 2024.06-dev that is running at the web location https://pirati.ca. The database version is 1561/1561, the post update version is 1550/1550.

**Comment by annando** [comment]
@foss- that specific host issue is caused by the **P** in `Pertsch.social`. By now we assumed that people will always use lowercase for hostnames.

**Comment by mexon** [comment]
I upgraded to 2024.06-dev and confirmed that the problem with `vote@fedi.vision` is still there, but is still fixed by my workaround.  Also that connecting to `@Sebastian@Pertsch.social` now works fine.  Trying to add `admin@takahe.social` results in a PHP error "Invalid contact." not an RSS feed as I expected.  Adding another workaround for that domain then allows me to follow that account as well.

**Comment by foss-** [comment]
Just followed `vote@fedi.vision` without issues. 
<img src="https://github.com/friendica/friendica/assets/5708172/d5dd2d8e-0f66-4f04-85a8-9c640618efa0" width="450">

Trying to follow `admin@takahe.social` resulted in `Not Found: Contact not found. Request: 6653a4248e830`.

This is Friendica, version 2024.06-dev that is running at the web location https://pirati.ca. The database version is 1563/1563, the post update version is 1550/1550.

**Comment by mexon** [comment]
@foss- as you can see in your screenshot, you are following `vote@fedi.vision` as an RSS feed, not as an ActivityPub contact.  That's the same result as I get.

(Edited the description to clarify)
